tests/unittests/GraphOptzTest.cpp

```diff
@@ -2210,6 +2210,22 @@ TEST_F(GraphOptz, mergeBANodes) {
   EXPECT_EQ(countNodeKind(F_, Kinded::Kind::BatchedAddNodeKind), 1);
 }
 
+/// Check that TileNodes that tile something only once are eliminated.
+TEST_F(GraphOptz, eliminateNoopTile) {
+  auto *A = mod_.createPlaceholder(ElemKind::FloatTy, {3, 1, 2}, "A",
+                                   /*isTrainable=*/false);
+  auto *tile = F_->createTile("tile", A, /*tiles=*/1,
+                              /*axis=*/1);
+  auto *relu = F_->createRELU("relu", tile);
+  F_->createSave("save", relu);
+
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::TileNodeKind), 1);
+  ::glow::optimize(F_, CompilationMode::Infer);
+
+  // Check that the Tile node is eliminated.
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::TileNodeKind), 0);
+}
```

Comment:
I think this is good, but similar to @kumudb's new optimization I think we should check that the functions outputs are identical before and after the change.

Reply:
My change for that just landed, I'll rebase and push.